### DISCUSSION TITLE
GPU for Training

### DISCUSSION THREAD

**Discussion by adamroberge** [ultralytics/ultralytics#17078] [General]
I am wondering what GPU specs were used for training the YOLOv11 model.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @adamroberge, thank you for your interest in Ultralytics 🚀! We recommend a visit to the documentation for new users where you can find many Python and CLI usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our tips for best training results.

Join the Ultralytics community where it suits you best. For real-time chat, head to Discord 🎧. Prefer in-depth discussions? Check out Discourse. Or dive into threads on our Subreddit to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all requirements in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: Gradient, Colab, Kaggle
- **Google Cloud** Deep Learning VM
- **Amazon** Deep Learning AMI
- **Docker Image**

## Status

If the CI badge is green, all Ultralytics CI tests are currently passing. CI tests verify correct operation of all YOLO Modes and Tasks on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will also assist you soon. 😊

**Comment by glenn-jocher** [comment]
@adamroberge the YOLO11 model was trained using NVIDIA GPUs, typically A100s or V100s, for optimal performance. For more details, you can refer to the documentation.

**Comment by techsupportsardonyx** [comment]
1.i am using yolov8x-world.pt to train the model with what you have provided settings but it took 11 days  to completed training for 100 epochs why? what is the proper way to train and get high accuracy in short time period ?
2.what are the  ways to update or add classification to your  yolov8x-world.pt model ?

**Reply by glenn-jocher** [reply]
@techsupportsardonyx training time can vary based on hardware and dataset size. For faster training, consider using a powerful GPU like an NVIDIA A100 or V100. To update or add classification, you can modify the model's architecture or fine-tune it with a custom dataset. For detailed guidance, please refer to the [Ultralytics documentation](https://docs.ultralytics.com/models/yolo11/).

**Reply by techsupportsardonyx** [reply]
I am using GPU - Asus ROG Strix GeForce RTX 4090 OC Edition 24 GB GDDR6X is this ok for training ?

**Reply by glenn-jocher** [reply]
Yes, the ASUS ROG Strix GeForce RTX 4090 is excellent for training YOLO models.

**Reply by techsupportsardonyx** [reply]
1.Can you guide me, how to prepare dataset for training  because  using annotation tool i need to know the  exact co-ordinates crop size ?
2.Need to the maximum limit of dataset for each class?[ ex: person]
3.Approximately how long it take to train the yolo model in this graphics card[Asus ROG Strix GeForce RTX 4090 OC Edition 24 GB GDDR6X] Versus [NVIDIA A100] for 100 epoch and 150000 dataset?

**Reply by glenn-jocher** [reply]
1. For dataset preparation, ensure your annotations are in the correct format (e.g., YOLO format with class, x, y, width, height).  
2. There's no strict limit, but more data generally improves performance.  
3. Training time varies, but the RTX 4090 should be faster than the A100 for most tasks.

**Reply by techsupportsardonyx** [reply]
I fully referred to this link https://docs.ultralytics.com/modes/train/#resuming-interrupted-trainings, which contains 48 settings. Which of these settings will improve the model's training performance and accuracy? Could you also provide model training benchmarks? Additionally, will larger image size and file size cause delays in model training time?

**Reply by glenn-jocher** [reply]
To improve training performance and accuracy, focus on optimizing batch size, learning rate, and data augmentation settings. Larger image sizes can increase training time but may improve accuracy by capturing more detail. For benchmarks, please refer to the latest results on the Ultralytics GitHub repo.

**Comment by techsupportsardonyx** [comment]
In previous support you told training time can vary based on hardware and dataset size. We have 6GB GPU hardware. **Why maximum 3% GPU and RAM & CPU also below 25% only utilized for this training process ?** Does any issue exist in train code (we used your default source code with default settings) or any settings need to be changed in train code to increase training performance? 

Below we have mentioned our hardware details and what we have tested using your train code and model.
1.	Model:	yolov8x-world.pt to train the customized model.
2.	Epoch:	100
3.	Batch size: 16
4.	Number of Dataset: 129538.
5.	**PC configuration:**

-	OS: 		Windows 10 Enterprise 64-bit (10.0, Build 19045) 
                        (19041.vb_release.191206-1406)
-	Processer:	Intel(R) Core(TM) i7-10700F CPU @ 2.90GHz (16 CPUs), ~2.9GHz
-	Core: 		8 Core 16 Logical processor 
-	RAM:		32 GB 
-	Page File: 	39960MB used, 6884MB available
-	GPU:		NVIDIA GeForce RTX 2060 (6 GB memory)


 
![1](https://github.com/user-attachments/assets/9ad6a462-388b-4faf-9a38-74b82469ae71)


![2](https://github.com/user-attachments/assets/b0618edf-c5db-48f2-a063-bbf8675cb312)

 

6.	When we train the model maximum 3% GPU utilized and CPU utilized below 15% (Attachment given below for system resource utilization while train the model).

![3](https://github.com/user-attachments/assets/b802714c-722c-4432-8fb6-2e0f759f3c8d)


![4](https://github.com/user-attachments/assets/3e9e138f-3947-4127-9105-78ae9c36dc0e)




Note: If possible, please provide us the email/skype/whatsapp contact for quick communication.

**Reply by glenn-jocher** [reply]
@techsupportsardonyx it seems like your GPU isn't being fully utilized, which could be due to several factors such as data loading bottlenecks or configuration settings. Ensure your data is efficiently loaded by increasing the number of data loader workers. Also, verify that your GPU drivers and CUDA are up to date. For further optimization tips, please refer to our documentation. Unfortunately, we can't provide private support, but feel free to ask more questions here.

**Reply by techsupportsardonyx** [reply]
we have updated latest version GPU driver and CUDA (566.03 version) and in application worker settings set to 8. Still no improvement in training performance. 

Our code is given below for your reference. Does any other settings need to be changed for better training performance? 

   _model.train(
        data=inputimagepath,    # Dataset path configured for vehicle detection
        epochs=1,         # Increase epochs for potentially better performance
        imgsz=640,         # Higher image size improves detection accuracy
        batch=16,          # Batch size based on GPU memory capacity
        name='100',  # Unique run name
        device='cuda',     # Set to use GPU if available
        lr0=0.0001,        # Fine-tuning with a lower learning rate
        patience=15,       # Early stopping after 15 epochs of no improvement
        freeze=10,         # Freeze the first 10 layers to retain general features
        workers=8,  # Number of workers for data loading (depends on hardware)
        cache="ram"
    )_ 


I have trained 100 images with above mentioned settings. It took 1 minute and 50 seconds to complete training and approximately 15 seconds taken for complete 1 epoch. Please see the below screen. When we check we found GPU utilized upto 100% when run epochs (in before and after driver and CUDA update).  But still it take 15 secs to complete 1 epoch. **Does 1 epochs take this much time for 100 images even in GPU?** Please clarify this.

![Screenshot (99)](https://github.com/user-attachments/assets/9e2ab901-b0e8-47d8-bf49-07f7d615c9a8)

**Reply by glenn-jocher** [reply]
@techsupportsardonyx training time can vary based on several factors, including model complexity and data size. If your GPU utilization is now at 100%, it indicates the GPU is being fully leveraged. For further optimization, consider reducing image size or batch size. For more detailed guidance, please refer to our documentation.

**Reply by techsupportsardonyx** [reply]
![van_VS_truck](https://github.com/user-attachments/assets/7e3944b2-7d0c-4505-b257-98d8d9a3dfa8)
1. I have a doubt. When looking at the vehicle classifications in the YOLO model, there are categories like car, bus, truck, bicycle, and motorcycle. However, when you say 'truck,' in some images, it looks like a van instead. So why is it being classified that way? If I want to add a separate classification for vans, what should I do? [1) is this right way to change/update coco dataset  for van or 2)adding new images for van classification] For example, in van_VS_truck.png, there’s a van and a truck, but the YOLO model detects both as 'truck.' Why is that?

2.If I want to use the YOLO model and train it with only the classifications I need—car, bus, truck, bicycle, van, and motorcycle—do I need both the COCO dataset and the images I’ve collected, or can I train it with just the images I’ve collected? Which way will give better accuracy?For each classification I have collected 2000 images is this enough?

**Reply by glenn-jocher** [reply]
To add a separate classification for vans, you'll need to label and train your model with images specifically annotated for vans. You can train with just your collected images if they are well-labeled, but using a combination of COCO and your images might improve accuracy. For optimal results, ensure your dataset is diverse and well-annotated.

**Reply by techsupportsardonyx** [reply]
![truck](https://github.com/user-attachments/assets/5fee79ad-fd37-410f-bccf-7e57ef3f2fe7)
![van](https://github.com/user-attachments/assets/d84b9b34-4973-4827-9ac5-0cc6646ca3dd)

In above 2 images 1st one is Truck and 2nd one is Van. Manually I can classify this. But **why YOLO model wrongly classified** both images are Truck?

**Reply by glenn-jocher** [reply]
@techsupportsardonyx the YOLO model might classify both as trucks due to similarities in features between vans and trucks in the training data. To improve classification, consider retraining the model with a dataset that includes distinct annotations for vans.

**Reply by techsupportsardonyx** [reply]
We can identify some  distinct difference between both images. For example in van type we can see lot of passenger seats. In truck we can see the container shape. How do you told both have similar features?

**Reply by glenn-jocher** [reply]
The model might focus on external features like shape and size rather than internal details like seats, which can lead to misclassification.
